automerge-frontend/src/state_tree/resolved_path.rs

```diff
@@ -3,200 +3,303 @@ use std::{convert::TryInto, num::NonZeroU32};
 use automerge_protocol as amp;
 
 use super::{
-    focus::Focus, random_op_id, DiffApplicationResult, LocalOperationResult, MultiGrapheme,
-    MultiValue, NewValueRequest, StateTree, StateTreeChange, StateTreeComposite, StateTreeList,
-    StateTreeMap, StateTreeTable, StateTreeText, StateTreeValue,
+    random_op_id, LocalOperationResult, MultiGrapheme, MultiValue, NewValueRequest, StateTree,
+    StateTreeComposite, StateTreeValue,
 };
 use crate::{error, Cursor, Primitive, Value};
 
 #[derive(Debug)]
 pub struct ResolvedPath<'a> {
-    root: &'a StateTree,
-    pub(crate) target: Target,
-}
-
-pub enum Target {
-    Root(ResolvedRoot),
-    Map(ResolvedMap),
-    Table(ResolvedTable),
-    List(ResolvedList),
-    Text(ResolvedText),
-    Character(ResolvedChar),
-    Counter(ResolvedCounter),
-    Primitive(ResolvedPrimitive),
-}
-
-impl std::fmt::Debug for Target {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        match self {
-            Target::Map(maptarget) => write!(f, "MapTarget {:?}", maptarget.value.object_id),
-            Target::Root(_) => write!(f, "Root"),
-            Target::Table(tabletarget) => {
-                write!(f, "Table {:?}", tabletarget.value.object_id)
-            }
-            Target::List(listtarget) => write!(f, "list {:?}", listtarget.value.object_id),
-            Target::Text(texttarget) => write!(f, "text {:?}", texttarget.value.object_id),
-            Target::Counter(countertarget) => write!(
-                f,
-                "counter {0}:{1:?}",
-                countertarget.containing_object_id, countertarget.key_in_container
-            ),
-            Target::Primitive(p) => write!(f, "primitive: {:?}", p.multivalue),
-            Target::Character(ctarget) => write!(f, "character {:?}", ctarget.multivalue),
-        }
-    }
+    pub(crate) target: Target<'a>,
```

Comment:
I think now that we don't need the  `root` to be carried around we could probably rename `Target` to `ResolvedPath` and remove the `Target` indirection entirely.

Reply:
Yep, removed Target and TargetMut now